eclair-core/src/main/scala/fr/acinq/eclair/db/sqlite/SqliteUtils.scala

```diff
@@ -99,7 +99,7 @@ object SqliteUtils {
    *
    * The lock will be kept until the database is closed, or if the locking mode is explicitly reset.
    */
-  def obtainExclusiveLock(sqlite: Connection) {
+  def obtainExclusiveLock(sqlite: Connection) = synchronized {
```

Comment:
This only fixes the issue if two calls to `obtainExclusiveLock` happen in the same process, but in the case of a background job running concurrently with the wallet app, this won't fix anything, right?
It's very surprising that sqlite doesn't have a way to atomically get a real exclusive lock or fail...

Reply:
This scenario is taken care of by the front application and should not happen. There's a logic that prevents background jobs to run concurrently with the front.

Reply:
Ok in that case the `synchronized` statement should work (and it doesn't hurt to have it anyway).
I still think we can probably do better on the sqlite side with a better understanding of their locking mechanisms, but it can be done later.